lib/compiler/src/sys_core_fold.erl

```diff
@@ -113,9 +115,34 @@ module(#c_module{defs=Ds0}=Mod, Opts) ->
 	_ -> ok
     end,
     init_warnings(),
+
+    %% Collect atoms which are optimized away
+    _Tid = ets:new(?OPT_ATOMS_TAB, [named_table, protected]),
```

Comment:
A named table will prevent running several instances of the compiler at the same time. I suggest that you use a `cerl_sets` that you put in the process dictionary.